### PR TITLE
UXENG-3184: Add bitbucket and gitlab scm docs to scm page

### PR DESCRIPTION
**Pull request by warmbowski**
Ticket: https://algorithmia.atlassian.net/browse/UXENG-3184

This PR adds documentation to the dev-center for Bitbucket and GitLab SCM-related functionality. Both features follow the existing oAuth flow for the existing GitHub SCM functionality.

### CODE REVIEW

**Comment by warmbowski** [comment]
I know that this looks like copy paste, but I went through a lot of SCM specific materials to make sure to get appropriate links and such. I am not very experienced with Bitbucket or Github, so please keep an eye out for SCM specific functionality that might be slightly different, or might be differently named. 

I will say, with a little more time and the right treatment, these three sections can probably generalized down to one, with appropriate screenshots that apply to all three. Just a suggestion though.

File: package.json
Comment (warmbowski) [inline package.json:11]: I removed rvm from command because I am using rbenv. Also, the version called out was old.

**Comment by warmbowski** [comment]
@StephanieKim I assume this shouldn't merge until the rest of the code merges into their main branch.

File: _pages/algorithm-development/source-control-management.md
Comment (bnlb) [inline _pages/algorithm-development/source-control-management.md]:
```suggestion
<img src="{{site.cdnurl}}{{site.baseurl}}/images/post_images/source_code_management/scm_create_bitbucket_algorithm_unauthorized.png" alt="Authorization prompt for Bitbucket repository host" class="screenshot img-sm">
```

File: _pages/algorithm-development/source-control-management.md
Comment (bnlb) [inline _pages/algorithm-development/source-control-management.md]:
```suggestion
- **An Access Key**: Bitbucket access keys allow read-only access to specific repositories, and are Bitbucket’s prescribed means by which external services can fetch code for building and deploying. These keys are not tied to individual permissions, and as such will allow Algorithmia to continue building an algorithm even if the permissions of the creating user change. We also use access keys to obtain your repository's commit log (to display changes when publishing versions) and README.md (for use as algorithm documentation).
```

File: _pages/algorithm-development/source-control-management.md
Comment (bnlb) [inline _pages/algorithm-development/source-control-management.md]:
```suggestion
If your algorithm's Bitbucket repository was deleted, Bitbucket does not have a mechanism for restoring. It is essential that both the repository's access key and webhooks remain in place, otherwise we will not be able to pull your source code, or know when you make changes.
```

File: _pages/algorithm-development/source-control-management.md
Comment (bnlb) [inline _pages/algorithm-development/source-control-management.md]: The endpoints for loading SCM data weren't returning secrets when I was testing Bitbucket endpoints. I'd leave that part off for now. Looking in BitBucket's UI I didn't see secrets mentioned anywhere.
```suggestion
Once you've found this value, visit your algorithm's repository on Bitbucket and click on the Settings tab. Click 'Webhooks' on the left side, and you'll be able to restore your webhook by entering the webhook URL.
```

File: _pages/algorithm-development/source-control-management.md
Comment (bnlb) [inline _pages/algorithm-development/source-control-management.md]:
```suggestion
We depend on webhooks to notify us when changes have occurred to your algorithm's repository. To restore your webhook within Bitbucket, first locate the webhook URL. This can be found on your algorithm's settings page by clicking 'View Details' under 'Bitbucket Webhook'.
```

File: _pages/algorithm-development/source-control-management.md
Comment (bnlb) [inline _pages/algorithm-development/source-control-management.md]:
```suggestion
You will then be presented with a modal, from which you can copy the webhook URL.
```

File: _pages/algorithm-development/source-control-management.md
Comment (bnlb) [inline _pages/algorithm-development/source-control-management.md]:
```suggestion
When you revoke access to an OAuth app, Bitbucket automatically revokes both your token and any access keys you may have created via the OAuth app. To fix your algorithms, simply [follow the instructions above](#your-repositorys-access-key-was-removed) on restoring deploy keys to a repository.
```

File: _pages/algorithm-development/source-control-management.md
Comment (bnlb) [inline _pages/algorithm-development/source-control-management.md]:
```suggestion
You can view your algorithm's builds by heading to its landing page and clicking the "Builds" tab if you are the algorithm's owner.
```

File: _pages/algorithm-development/source-control-management.md
Comment (bnlb) [inline _pages/algorithm-development/source-control-management.md]:
```suggestion
- **A Deploy Key**: GitLab deploy keys allow read-only access to specific repositories, and are GitLab’s prescribed means by which external services can fetch code for building and deploying. These keys are not tied to individual permissions, and as such will allow Algorithmia to continue building an algorithm even if the permissions of the creating user change. We also use deploy keys to obtain your repository's commit log (to display changes when publishing versions) and README.md (for use as algorithm documentation).
```

**Review by bnlb — APPROVED** [review]
lgtm!

**Review by stevielkim — APPROVED** [review]
Looks great! Thanks @bnlb for the awesome first review as well!

**Comment by stevielkim** [comment]
@warmbowski hey, sorry, but I've learned that there is both a BitBucket server and BitBucket cloud implementation - BitBucket Cloud was added without my awareness. Were these directions for BB server or cloud? We'll need docs for both and to call out which is which by their names "BitBucket Server" and "BitBucket Cloud"

**Comment by warmbowski** [comment]
This is all Bit Bucket cloud. There is no UI functionality for BB server.

**Pull request merged**
